### PR TITLE
Be stricter about export targets and build configurations

### PR DESCRIPTION
**Pull request by McMartin**
This PR fixes a bug and makes some ground work for upcoming feature PRs on `Reprojucer.cmake`.

@MartyLake: please review, thanks!

### CODE REVIEW

File: ci/CMakeLists.txt [hunk 1]
```diff
@@ -37,14 +37,18 @@ add_subdirectory("../Jucer2Reprojucer/generated/JUCE/examples/audio plugin host"
   "Plugin Host"
 )
 
+if(NOT MSVC_VERSION EQUAL 1900)
```
Comment (MartyLake) [inline ci/CMakeLists.txt]: How are these tests related to the commit message ?

Comment (McMartin) [inline ci/CMakeLists.txt]: `NetworkGraphicsDemo.jucer` doesn't have a Visual Studio 2015 exporter, so we cannot build it with `Reprojucer.cmake` when using Visual Studio 2015 (i.e. `MSVC_VERSION == 1900`).

**Comment by MartyLake** [comment]
I understand what this test means. But how is this related to the commit
message ? Shouldnt it be split to its own commit instead ?


---
Matthieu TALBOT
neogaldr@gmail.com

+33 629 22 15 70
fr.linkedin.com/in/matthieutalbot
http://www.soundcloud.com/Marty-Lake
http://lifehacker.com/5028808/how-eom-makes-your-email-more-efficient

**Comment by McMartin** [comment]
fa203dd makes an API breaking change, since it was not required to call `jucer_export_target()` with the current exporter before, thus we have to adapt the clients to the API breaking change in that commit.

**Comment by MartyLake** [comment]
ok!!!! LGTM then :)


---
Matthieu TALBOT
neogaldr@gmail.com

+33 629 22 15 70
fr.linkedin.com/in/matthieutalbot
http://www.soundcloud.com/Marty-Lake
http://lifehacker.com/5028808/how-eom-makes-your-email-more-efficient
